.github/workflows/test-notebooks_gen.yaml

```diff
@@ -0,0 +1,219 @@
+# THIS FILE IS GENERATED. PLEASE DO NOT MODIFY DIRECTLY.
+# Please refer to the `test_notebooks.py` file in the `featurebyte/infrastructure` repo if you want to update it.
+
+name: test-notebooks
+'on':
+  workflow_dispatch: {}
+  pull_request:
+    types:
+    - opened
+    - synchronize
+    - reopened
+  push:
+    branches:
+    - main
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+env:
+  GCR_DOCKER_LOCATION: us-central1
+  POETRY_HTTP_BASIC_FEATUREBYTE_NP_PASSWORD: ${{ secrets.GCR_PYPI_CREDENTIALS_R_B64 }}
+  POETRY_HTTP_BASIC_FEATUREBYTE_NP_USERNAME: _json_key_base64
+jobs:
+  test-deep-dive-notebooks:
+    runs-on: ubuntu-latest-4-cores
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: arduino/setup-task@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        version: 3.x
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Setup cache
+      uses: actions/cache@v3
+      with:
+        key: poetry-cache-${{ runner.os }}-${{ matrix.python-version }}-${{ github.job }}
+        path: .venv
+        restore-keys: poetry-cache-${{ runner.os }}-${{ matrix.python-version }}
+    - name: Install Poetry
+      uses: abatilo/actions-poetry@v2
+      with:
+        poetry-version: 1.5.1
+    - name: Configure Poetry
+      run: |-
+        poetry config virtualenvs.in-project true
+        poetry config http-basic.featurebyte_np ${POETRY_HTTP_BASIC_FEATUREBYTE_NP_USERNAME} ${POETRY_HTTP_BASIC_FEATUREBYTE_NP_PASSWORD}
+    - name: Install packages needed for build
+      run: sudo apt-get install libkrb5-dev libsasl2-dev libpython3-dev g++ gcc
+    - name: Install packages needed for Kerberos
```

Comment:
nit: this step seems unnecessary since there's it's already installed in the build.